### DISCUSSION TITLE
Print Map of On-Cluster Traffic Flow(s)

### DISCUSSION THREAD

**Discussion by todd-dsm** [istio/istio#53404] [General]
Need to understand how traffic flows within an inherited and moderately complex Istio config. 

Would there happen to be a way to print a hierarchical map of traffic flows from `istioctl` (or any other method) by chance? Any output format would be fine:
* ASCII
* `graphviz` 
* whatever

**Comment by howardjohn** [comment]
This is essentially what Kiali provides: https://kiali.io/.

**Reply by todd-dsm** [reply]
Kiali is in the browser and live-only. The question is about creating an image. The intent is specifically to help with architectural visualizations. Again, for an inherited system that isn't easy to understand. This, of course, can be done the hard way. But, since we're in the business of automation...

Does Kiali have the facility to drop an image?

**Reply by howardjohn** [reply]
I was going to discuss the browser aspect but you said 'any output format would be fine' so I dropped it :slightly_smiling_face: .

Regardless, I would think building out alternative visualization mechanism on top of/in kiali would be more beneficial. Building the graph is not a trivial problem, so building on that work would like be quite nice.

Or you could do some headless browser snapshot thing but that may not be great.

Kiali also has an API that you can probably query and translate to graphviz or w/e

**Comment by todd-dsm** [comment]
> Kiali also has an API that you can probably query and translate to graphviz or w/e

That sounds interesting; I'm not a dev so I'll have to reach out to them. But, now I know where to go - thank you 😀
